### PR TITLE
curvefs: add memcached startup check

### PR DESCRIPTION
**Pull request by Ziy1-Tan**
### What problem does this PR solve?

Issue Number: https://github.com/opencurve/curve/issues/2295

Problem Summary:

### What is changed and how it works?

What's Changed:

How it Works:

Side effects(Breaking backward compatibility? Performance regression?):

### Check List

- [ ] Relevant documentation/comments is changed or added
- [ ] I acknowledge that all my contributions will be made under the project's license

### CODE REVIEW

**Comment by Ziy1-Tan** [comment]
cicheck

**Comment by Ziy1-Tan** [comment]
cicheck

File: curvefs/test/client/client_memcache_test.cpp [hunk 1]
```diff
@@ -65,14 +65,21 @@ class MemCachedTest : public ::testing::Test {
...
                 break;
             }
             std::this_thread::sleep_for(std::chrono::milliseconds(100));
             retry++;
         } while (1);
-        LOG(INFO) << "=============== memcache start ok";
+        isRunning_ = ret && retry <= 100;
+        if (!isRunning_) {
+            LOG(ERROR) << absl::StrCat(
+                "=============== memcache start failed, errorlog: ", errorlog);
```
Comment (Cyber-SiKu) [inline curvefs/test/client/client_memcache_test.cpp:79]: Use `ASSERT_EQ` checks instead of  output.

Comment (Ziy1-Tan) [reply curvefs/test/client/client_memcache_test.cpp]: if Use `ASSERT_EQ`, The ci still failed. I use `isRunning_` to skip the following test.

**Comment by Ziy1-Tan** [comment]
It seems that it will fail when `memcached -u root 18080`
image

File: curvefs/test/client/client_memcache_test.cpp [hunk 2]
```diff
@@ -65,13 +65,16 @@ class MemCachedTest : public ::testing::Test {
...
+        bool ret = false;
         do {
-            if (client->Set("1", "2", 1, &errorlog) || retry > 100) {
+            if ((ret = client->Set("1", "2", 1, &errorlog)) || retry > 100) {
                 break;
             }
             std::this_thread::sleep_for(std::chrono::milliseconds(100));
             retry++;
         } while (1);
+        ASSERT_TRUE(ret && retry <= 0)
```
Comment (ilixiaocui) [inline curvefs/test/client/client_memcache_test.cpp:76]: ASSERT_TRUE(ret && retry <= 100)

**Comment by Ziy1-Tan** [comment]
cicheck

**Comment by Ziy1-Tan** [comment]
It seems that it must fail when `memcached -u root 18080` wit root user
image

**Comment by ilixiaocui** [comment]
The error is observed as follows:

image

When we use sudo to execute, the identified user is ` root`. There is an extra space before `root`.

I have tried following and it's ok.
```
 ASSERT_EQ(0, execlp("memcached", "memcached", "-uroot",
                                memcached_port.c_str(), nullptr));

```

**Comment by Ziy1-Tan** [comment]
cicheck

**Review by Cyber-SiKu — APPROVED** [review]

**Review by ilixiaocui — APPROVED** [review]

**Pull request merged**